src/EventListener/CacheControlListener.php

```diff
@@ -55,10 +64,12 @@ class CacheControlListener extends AbstractRuleListener implements EventSubscrib
     private $debugHeader;
 
     /**
-     * @param string|bool $debugHeader Header to set to trigger debugging, or false to send no header
+     * @param RuleMatcherInterface $cacheableMatcher Matcher to decide whether caching headers should be set
+     * @param string|bool          $debugHeader      Header to set to trigger debugging, or false to send no header
      */
-    public function __construct($debugHeader = false)
+    public function __construct(RuleMatcherInterface $cacheableMatcher, $debugHeader = false)
```

Comment:
we could avoid the BC break by using = null for the matcher and create a default one if none is passed. would that be the right thing to do or too many options?